File: 박예진/3주차/260113.cpp
//https://www.acmicpc.net/problem/28324

#include <iostream>
#include <vector>
using namespace std;

int N;
long long V[500001];

int main() {
  ios::sync_with_stdio(false);
  cin.tie(0); cout.tie(0);

  cin >> N;
  for(int i = 0; i < N; i++){
    cin >> V[i];
  }

  long long cnt = 1;
  long long ans = 1;
  for(int i = N - 2; i >= 0; i--){
    if (V[i] >= cnt + 1) cnt++; 
    else cnt = V[i];
    ans += cnt;
  }
  cout << ans;
  return 0;
}

File: 정건우/3주차/260113.java
//https://www.acmicpc.net/problem/14241
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Collections;
import java.util.PriorityQueue;
import java.util.StringTokenizer;

public class BOJ_S3_14241_슬라임합치기 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());

        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            pq.add(Integer.parseInt(st.nextToken()));
        }

        int ans = 0;

        while (pq.size() >= 2) {
            int x = pq.poll();
            int y = pq.poll();

            ans += x * y;
            pq.add(x+y);
        }

        System.out.println(ans);

    }
}
